### PR TITLE
Use simpleservo embedding API for Magic Leap

### LINKED ISSUE #22065: Integrate libsimpleservo into magic leap port
I was able to get a build which links with the following diff:
```diff
diff --git a/Cargo.lock b/Cargo.lock
index 1058bfdac3..77ce164f99 100644
--- a/Cargo.lock
+++ b/Cargo.lock
@@ -1906,6 +1906,7 @@ name = "libmlservo"
 version = "0.0.1"
 dependencies = [
  "libservo 0.0.1",
+ "libsimpleservo 0.0.1",
  "log 0.4.5 (registry+https://github.com/rust-lang/crates.io-index)",
  "servo-egl 0.2.1 (registry+https://github.com/rust-lang/crates.io-index)",
  "smallvec 0.6.3 (registry+https://github.com/rust-lang/crates.io-index)",
diff --git a/ports/libmlservo/Cargo.toml b/ports/libmlservo/Cargo.toml
index 793435db09..f66bfab03c 100644
--- a/ports/libmlservo/Cargo.toml
+++ b/ports/libmlservo/Cargo.toml
@@ -13,6 +13,7 @@ bench = false

 [dependencies]
 libservo = { path = "../../components/servo" }
+libsimpleservo = { path = "../libsimpleservo" }
 log = "0.4"
 servo-egl = "0.2"
 smallvec = "0.6"
diff --git a/ports/libmlservo/src/lib.rs b/ports/libmlservo/src/lib.rs
index 08b84ada37..9188dfffa9 100644
--- a/ports/libmlservo/src/lib.rs
+++ b/ports/libmlservo/src/lib.rs
@@ -3,6 +3,7 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

 extern crate egl;
+extern crate simpleservo;
 #[macro_use] extern crate log;
 extern crate servo;
 extern crate smallvec;
diff --git a/ports/libsimpleservo/Cargo.toml b/ports/libsimpleservo/Cargo.toml
index 5af57fe8e6..f942871baf 100644
--- a/ports/libsimpleservo/Cargo.toml
+++ b/ports/libsimpleservo/Cargo.toml
@@ -8,7 +8,7 @@ publish = false

 [lib]
 name = "simpleservo"
-crate-type = ["cdylib"]
+#crate-type = ["cdylib"]
 test = false
 bench = false

diff --git a/ports/libsimpleservo/build.rs b/ports/libsimpleservo/build.rs
index 0ba96d6b7a..60366c9c59 100644
--- a/ports/libsimpleservo/build.rs
+++ b/ports/libsimpleservo/build.rs
@@ -12,7 +12,7 @@ use std::path::Path;

 fn main() {
     let target = env::var("TARGET").unwrap();
-    if target.contains("android") {
+    if target.contains("android") && env::var("SERVO_FAKE_ANDROID").is_err() {
         android_main()
     }
     generate_gl_bindings(&target);
```
If libsimpleservo is a cdylib, libmlservo can't find it because the rustc commandline doesn't include an `--extern simpleservo=...` line.

### PR DESCRIPTION
**Pull request by jdm**
This removes the duplication between the two ports and makes it easier to improve all of our embeddings simultaneously.

---
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #22065
- [x] These changes do not require tests because no automated tests for embedded devices.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/22871)

### CODE REVIEW

**Comment by jdm** [comment]
r? @asajeffrey 
Paul already signed off on the non-ML changes in https://github.com/servo/servo/issues/22065#issuecomment-457817409.

**Comment by asajeffrey** [comment]
My hot take is that this is much nicer. I'll have a less-hot take tomorrow.

File: ports/libmlservo/src/lib.rs [hunk 1]
```diff
@@ -101,6 +84,16 @@ pub struct MLApp(*mut c_void);
 
 const LOG_LEVEL: log::LevelFilter = log::LevelFilter::Info;
 
+fn call<F>(f: F) -> Result<(), &'static str>
```
Comment (asajeffrey) [inline ports/libmlservo/src/lib.rs]: Parameterize this over the return type?

File: ports/libmlservo/src/lib.rs [hunk 2]
```diff
@@ -109,150 +102,73 @@ pub unsafe extern "C" fn init_servo(
...
-    servo::embedder_traits::resources::set(Box::new(ResourceReaderInstance::new()));
     let _ = log::set_boxed_logger(Box::new(logger));
     log::set_max_level(LOG_LEVEL);
-    let gl = GlesFns::load_with(|symbol| {
-        let cstr = CString::new(symbol).expect("Failed to convert GL symbol to a char*");
-        eglGetProcAddress(cstr.as_ptr() as _) as _
-    });
 
+    let gl = gl_glue::egl::init().expect("EGL initialization failure");
     info!("OpenGL version {}", gl.get_string(gl::VERSION));
```
Comment (asajeffrey) [inline ports/libmlservo/src/lib.rs]: I think we can get rid of this `info!`, it was useful while debugging but not any more.

File: ports/libmlservo/src/lib.rs [hunk 3]
```diff
@@ -101,6 +84,16 @@ pub struct MLApp(*mut c_void);
 
 const LOG_LEVEL: log::LevelFilter = log::LevelFilter::Info;
 
+fn call<F>(f: F) -> Result<(), &'static str>
+where
+    F: FnOnce(&mut ServoGlue) -> Result<(), &'static str>,
+{
+    SERVO.with(|s| match s.borrow_mut().as_mut() {
```
Comment (asajeffrey) [inline ports/libmlservo/src/lib.rs]: Sigh, `libsimpleservo` uses a global variable? This is a bit of a step back, `libmlservo` passed the servo instance around as a parameter.

**Review by asajeffrey — APPROVED** [review]
OK, this looks much much nicer. The use of the `SERVO` global variable is not what I would do, but everything else is much nicer.

File: ports/libmlservo/src/lib.rs (hunk 3 above)
Comment (jdm) [reply ports/libmlservo/src/lib.rs]: I'll file a followup issue to discuss this choice.

**Comment by jdm** [comment]
@bors-servo retry
* #21067

**Pull request merged**